lib/datasource/index.ts

```diff
@@ -124,3 +126,45 @@ export function getDigest(
     value
   );
 }
+export function sanitizeSourceUrl(rawSourceUrl){
```

Comment:
We can perhaps use a "home made" function like this as a fallback, but I prefer to use https://github.com/npm/hosted-git-info as the first option because I know it is battle-tested.

Reply:
@rarkins it is not all that complicated parsing-wise, but as you wish.

Reply:
Marking this as unresolved